api/v1alpha1/webhook.go

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+)
+
+// Maximum length of bundle name
+const maxNameLength = 40
+
+// log is for logging in this package.
+var bundlelog = logf.Log.WithName("bundle-resource")
+
+func (r *Bundle) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	bundlelog.Info("SetupWebhookWithManager")
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(r).
+		Complete()
+}
+
+//+kubebuilder:webhook:path=/validate-core-rukpak-io-v1alpha1-bundle,mutating=false,failurePolicy=fail,sideEffects=None,groups=core.rukpak.io,resources=bundles,verbs=create;update,versions=v1alpha1,name=core.rukpak.io,admissionReviewVersions=v1
+
+var _ webhook.Validator = &Bundle{}
+
+// ValidateCreate implements webhook.Validator so a webhook will be registered for the type
+func (r *Bundle) ValidateCreate() error {
+	bundlelog.Info("validate create", "name", r.Name)
+
+	return checkNameLength(r)
```

Comment:
It might be helpful to emit a log in the case where the name check returns an error, WDYT? Something like:

```
err := checkNameLength(r)
if err != nil {
  bundlelog.Infof("validating bundle create: %s")
} 
return nil
```

Reply:
I think it's generating enough information to the console and in the log now.

Console output
```
kubectl apply -f bundle_longname.yaml 
Error from server (bundle name combo-78901234567890123456789012345678901 is too long: maximum allowed name length is 40): error when creating "bundle_longname.yaml": admission webhook "bundle-rukpak-webhook.rukpak-system.svc" denied the request: bundle name combo-78901234567890123456789012345678901 is too long: maximum allowed name length is 40
```

Log entry
```
1.647888685788609e+09   DEBUG   bundle-resource validate create {"name": "combo-78901234567890123456789012345678901"}
1.647888685788845e+09   DEBUG   controller-runtime.webhook.webhooks     wrote response  {"webhook": "/validate-core-rukpak-io-v1alpha1-bundle", "code": 403, "reason": "bundle name combo-78901234567890123456789012345678901 is too long: maximum allowed name length is 40", "UID": "f238221b-e3f3-4b1c-a4e9-370f5cf05aa4", "allowed": false}
```
Are these enough or need some more?